pkg/pillar/types/volumetypes.go

```diff
@@ -113,6 +113,8 @@ type VolumeStatus struct {
 	CurrentSize             int64  // current total downloaded size as reported by the downloader
 	FileLocation            string // Location of filestystem
 	VolumeCreated           bool   // Done aka Activated
+	VolumePrepareDone       bool
+	VolumePreparing         bool
```

Comment:
I think it makes sense to replace these three booleans by a some new VolumeSubState enum. I think that makes the logic simpler (avoid checking three booleans in the if statements)

Reply:
Done. Also get rid of VolumeCreated in favor of substate.

Reply:
Thanks